client/packages/programs/src/JsonForms/components/ProgramEvent.tsx

```diff
@@ -15,23 +15,29 @@ import { EncounterFragment, useEncounter, useProgramEvents } from '../../api';
 
 import { z } from 'zod';
 
-export const encounterProgramEventTester = rankWith(
-  10,
-  uiTypeIs('EncounterProgramEvent')
-);
+export const programEventTester = rankWith(10, uiTypeIs('ProgramEvent'));
 
+/**
+ * Program events work with specific encounters, or generically
+ * (by providing both a documentType and eventType). Encounters still
+ * need to be fetched from the api since some encounters
+ * require events to be filtered out by the encounter start datetime.
+ */
 const Options = z
   .object({
     /**
-     * Time of the program event:
+     * This option should only be configured for encounters.
```

Comment:
```suggestion
     * This option should only be configured if the current document is an encounter.
```

Just noticed, what about `at` times for non-encounter documents. We don't want to always show events at `at: now()`...

Maybe better have a config like:

```ts
{
  /** The time when program active where active */
  at: {
     encounterStartDatetime: z.enum(['before', 'after']).optional(),
     programEnrolmentDatetime: z.enum(['before', 'after']).optional(),
     documentDatetime: z.enum(['before', 'after']).optional(),
  }
}
```